src/django_rich/runner.py

```diff
@@ -0,0 +1,119 @@
+        if self.showAll:
+            self.console.print("FAIL", style=Style(color="red"))
+        elif self.dots:
+            self.console.print("F", style=Style(color="red"), end="")
+
+    def addSkip(self, test, reason):
+        super(unittest.TextTestResult, self).addSkip(test, reason)
+        if self.showAll:
+            self.console.print(f"skipped {reason!r}", style=Style(color="yellow"))
+        elif self.dots:
+            self.console.print("s", style=Style(color="yellow"), end="")
+
+    def addExpectedFailure(self, test, err):
+        super(unittest.TextTestResult, self).addExpectedFailure(test, err)
+        if self.showAll:
+            self.console.print("expected failure", style=Style(color="yellow"))
+        elif self.dots:
+            self.console.print("x", style=Style(color="yellow"), end="")
+
+    def addUnexpectedSuccess(self, test):
+        super(unittest.TextTestResult, self).addUnexpectedSuccess(test)
+        if self.showAll:
+            self.console.print("unexpected success", style=Style(color="red"))
+        elif self.dots:
+            self.console.print("u", style=Style(color="red"), end="")
+
+    def _exc_info_to_string(self, err, test):
+        """Converts a sys.exc_info()-style tuple of values into a string."""
+        exctype, value, tb = err
+        # Skip test runner traceback levels
+        while tb and self._is_relevant_tb_level(tb):
+            tb = tb.tb_next
+
+        extract = Traceback.extract(exctype, value, tb, show_locals=True)
+        tb_e = Traceback(extract, suppress=[unittest, testcases])
+        with self.console.capture() as capture:
+            self.console.print(tb_e)
+        msgLines = capture.get()
+        if self.buffer:
+            output = sys.stdout.getvalue()
+            error = sys.stderr.getvalue()
+            if output:
+                if not output.endswith("\n"):
+                    output += "\n"
+                msgLines.append(STDOUT_LINE % output)
+            if error:
+                if not error.endswith("\n"):
+                    error += "\n"
+                msgLines.append(STDERR_LINE % error)
+        return "".join(msgLines)
+
+    def printErrorList(self, flavour, errors):
+        for test, err in errors:
+            r = Rule(style=self.django_green)
+            title = f"{flavour}: {self.getDescription(test)}"
+            self.console.print(r, title, r)
+            self.stream.writeln("%s" % err)
+
+
+class RichTextTestResult(RichMixin, unittest.TextTestResult):
+    pass
+
+
+class RichDebugSQLTextTestResult(RichMixin, DebugSQLTextTestResult):
+    pass
+
+
+class RichPDBDebugResult(RichMixin, PDBDebugResult):
+    pass
+
+
+class RichTestRunner(unittest.TextTestRunner):
+    resultclass = RichTextTestResult
+
+
+class RichDiscoverRunner(DiscoverRunner):
+    test_runner = RichTestRunner
+
+    def get_resultclass(self):
+        if self.debug_sql:
+            return RichDebugSQLTextTestResult
```

Comment:
You mentioned about `--pdb` and `--debug-sql` on twitter. This covers those off, anything else that I'm missing (very likely).

Reply:
Nice one.